test/test_view_ops.py

```diff
@@ -1038,6 +1038,34 @@ def test_broadcast_shapes(self, device):
                 actual = torch.broadcast_shapes(s0, s1)
                 self.assertEqual(expected, actual)
 
+    # Skip BFloat16 since numpy does not support it
+    @dtypes(*torch.testing.get_all_dtypes(include_bfloat16=False))
+    def test_broadcast_to(self, device, dtype):
```

Comment:
There's one quirk of torch.Tensor.expand that's different from NumPy's broadcast_to. NumPy's broadcast_to won't accept -1 as a value, but PyTorch's expand will.

NumPy:
```
a = np.array((1, 2, 3))
np.broadcast_to(a, (-1,))
: ValueError: all elements of broadcast shape must be non-negative
```

PyTorch:
```
t = torch.tensor((1, 2, 3))
t.expand(-1)
: tensor([1, 2, 3])
```

I think this discrepancy is OK, however, as this just adds functionality to NumPy's broadcast_to.

Reply:
Thanks for the detailed illustration regarding the discrepancy between NumPy's broadcast_to and `torch.broadcast_to` function.
The check for `non-negative` shape is not added for `torch.broadcast_to`, making it behave consistently with `torch.Tensor.expand`.